DatadogSessionReplay/Sources/DatadogSessionReplay/Recorder/Utilities/ImageDataProvider.swift

```diff
@@ -0,0 +1,98 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import UIKit
+
+internal class ImageDataProvider {
+    enum DataLoadingStatus: Encodable {
+        case loading, loaded(_ base64: String), ignored
+    }
+
+    private var cache: Cache<String, DataLoadingStatus>
+    private var queue: Queue
+    private let maxBytesSize: Int
+    private let maxDimensions: CGSize
+
+    internal init(
+        cache: Cache<String, DataLoadingStatus> = .init(),
+        queue: Queue = BackgroundAsyncQueue(named: "com.datadoghq.session-replay.image-data-provider"),
+        maxBytesSize: Int = 64_000,
+        maxDimensions: CGSize = CGSize(width: 120, height: 120)
+    ) {
+        self.cache = cache
+        self.queue = queue
+        self.maxBytesSize = maxBytesSize
+        self.maxDimensions = maxDimensions
+    }
+
+    func contentBase64String(of imageView: UIImageView) -> String? {
+        guard var image = imageView.image else {
+            return ""
+        }
+
+        var identifier: String
+        if let name = image.name {
+            identifier = name
+        } else {
+            identifier = "\(image.hash)"
+        }
+
+        let tintColor = imageView.tintColor
+        if let tintColor = tintColor {
+            identifier += "\(hexStringFromColor(color: tintColor))"
```

Comment:
The cache we added is based on generics 👌 (`Cache<Key: Hashable, Value>`), hence we don't need to slow down performance by turning `UIColor` into `String`. We can use `UIColor` (hashable) as a dimention of the cache key.

Reply:
I initially used hash, but it wasn't sufficient for testing (it's NSObject hash with is memory address).